src/mobile/src/ui/components/Migration.js

```diff
@@ -59,18 +73,37 @@ class Migration extends Component {
         startTrackingProgress: PropTypes.func.isRequired,
         /** @ignore */
         completedMigration: PropTypes.bool.isRequired,
+        /** @ignore */
+        notificationLog: PropTypes.array.isRequired,
+        /** @ignore */
+        nodes: PropTypes.array.isRequired,
+        /** @ignore */
+        setFullNode: PropTypes.func.isRequired,
+        /** @ignore */
+        isChangingNode: PropTypes.bool.isRequired,
     };
 
+    constructor() {
+        super();
+        this.state = {
+            hasFailedMigration: false,
+        };
+        this.changeNode = this.changeNode.bind(this);
+        this.retryMigration = this.retryMigration.bind(this);
+    }
+
     componentDidMount() {
         this.props.startTrackingProgress(ProgressSteps.migration);
-
         this.props.migrate(reduxPersistStorageAdapter);
     }
 
     componentWillReceiveProps(newProps) {
         if (!this.props.completedMigration && newProps.completedMigration) {
             this.navigateToLoadingScreen();
         }
+        if (this.props.notificationLog !== newProps.notificationLog) {
```

Comment:
`notificationLog` is an array. Perhaps we should be comparing size here (Also considering that there could be previous error logs).

Reply:
Would this trigger if the user cleared the error logs?